blockchain/blockchain.go

```diff
@@ -224,7 +227,7 @@ type prefetchTx struct {
 // NewBlockChain returns a fully initialised block chain using information
 // available in the database. It initialises the default Klaytn validator and
 // Processor.
-func NewBlockChain(db database.DBManager, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine, vmConfig vm.Config) (*BlockChain, error) {
+func NewBlockChain(db database.DBManager, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine, vmConfig vm.Config, livePruningEnabled bool) (*BlockChain, error) {
```

Comment:
Instead of introducing a new parameter, how about introducing a setter?
```go
bc := NewBlockChain(~~)
bc.EnableLivePruning()
```

Reply:
Please pass it via `CacheConfig.LivePruningEnabled`

Reply:
Iniitally I pondered @blukat29's suggestion, but the flag doesn't quite match the context of cache configuration. I find @ian0371's suggestion appealing. What's your take on it?

Reply:
I find CacheConfig (except for its name "cache") appropriate to put LivePruningEnabled flag, because it also contains other Trie-related parameters like ArchiveMode, TriesInMemory, LivePruningRetention.

Reply:
Moved into `CacheConfig`. PTAL.